index.js

```diff
@@ -102,3 +49,67 @@ exports.fromJSON = fromJSON;
 function fromJSON(data) {
   return reader.read(data);
 }
+
+exports.createWriter = createWriter;
```

Comment:
Instead of exporting `createWriter`, could you export something like:

```js
function withFilter(predicate) {
  var filteredWriter = createWriter(predicate);
  return {
    toJSON: function(data) {
      return filteredWriter.write(data);
    },
    fromJSON: fromJSON
  }
}
```